src/error.rs

```diff
@@ -4,39 +4,27 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use core::fmt;
-
-#[cfg(feature = "std")]
-use std::{error as std_error, fmt as std_fmt};
+use core::fmt::{self, Display, Result};
 
 /// Poseidon error variants
-#[derive(Copy, Clone, Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
-pub enum Error<E: fmt::Debug> {
+#[derive(Clone, Debug)]
+pub enum Error {
     /// Error pushing to the poseidon tree
-    TreePushFailed(E),
+    TreePushFailed,
     /// Error on pop of the tree
-    TreePopFailed(E),
+    TreePopFailed,
     /// Error fetching the Nth item from the tree
-    TreeGetFailed(E),
+    TreeGetFailed,
+    /// Failed to obtain a Branch from a tree.
+    TreeBranchFailed,
+    /// Failed to obtain an Iterator from a tree.
+    TreeIterFailed,
     /// Decryption failed for the provided secret+nonce
     CipherDecryptionFailed,
 }
```

Comment:
Wondering why this peculiar naming convention: why add `Failed` at suffix? It's not very idiomatic. Not suggesting to change now at all (it would be a MASSIVE breaking change) it just looks a bit weird. Maybe we could change before the release `1.0`?

Reply:
Filed https://github.com/dusk-network/Poseidon252/issues/140